frontend/public/components/secrets/create-secret/PullSecretUploadForm.tsx

```diff
@@ -0,0 +1,54 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { DroppableFileInput, PullSecretData } from '.';
+
+export const PullSecretUploadForm: React.FC<PullSecretUploadFormProps> = ({
+  onChange,
+  onDisable,
+  stringData,
+}) => {
+  const { t } = useTranslation();
+  const [configFile, setConfigFile] = React.useState<string>(
+    _.isEmpty(stringData) ? '' : JSON.stringify(stringData),
+  );
```

Comment:
This is a controlled component, so we shouldn't need state:
```suggestion
  const configFile = React.useMemo<string>(() =>
    pullSecretData ? JSON.stringify(pullSecretData) : '',
    [pullSecretData],
  );
```